src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,151 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> node = new Node<>(tail, value, null);
+        if (tail == null) {
+            head = node;
+        } else {
+            tail.next = node;
+            node.prev = tail;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        Node<T> oldNode = getNode(index);
+        Node<T> node = new Node<>(oldNode.prev, value, oldNode);
+        if (oldNode.prev == null) {
+            head = node;
+        } else {
+            oldNode.prev.next = node;
+        }
+        oldNode.prev = node;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T nodeItem : list) {
+            add(nodeItem);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNode(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        T result;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        result = node.value;
+        node.value = value;
+        return result;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> node = getNode(index);
+        unlink(node);
+        size--;
+        return node.value;
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        int index = getIndex(object);
+        if (index == -1) {
+            return false;
+        }
+        remove(index);
+        return true;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void unlink(Node<T> node) {
+        if (node.prev == null) {
+            head = node.next;
+        } else {
+            node.prev.next = node.next;
+        }
+        if (node.next == null) {
+            tail = node.prev;
+        } else {
+            node.next.prev = node.prev;
+        }
+    }
+
+    private int getIndex(T value) {
+        Node<T> node = head;
+        int i = 0;
+        do {
+            T valueNode = node.value;
+            if (valueNode != null && valueNode.equals(value) || valueNode == value) {
+                return i;
+            }
+            i++;
+            node = node.next;
+        } while (node.next != null);
+        return -1;
```

Comment:
let's use `while` loop here not `do while`) Also, let's not add separate method for this logic and move to `remove(T value)` method and we don't need to user `remove()`by index method in `remove(T value)`)